task.h

```diff
@@ -68,4 +86,45 @@ void print_task(task_spec *spec, UT_string *output);
 /* Parse task as printed by print_task. */
 task_spec *parse_task(char *task_string, int64_t task_length);
 
+/*
+ * SCHEDULED TASK: Contains information about a scheduled task:
+ * the task iid, the task specification and the task status
+ * (WAITING, SCHEDULED, RUNNING, DONE) and which node the
+ * task is scheduled on.
+ *
+ */
+
+/* The scheduling_state can be used as a flag when we are listening for an event,
+ * for example TASK_WAITING | TASK_SCHEDULED. */
+enum scheduling_state {
+  TASK_WAITING = 0,
+  TASK_SCHEDULED = 1,
+  TASK_RUNNING = 2,
+  TASK_DONE = 4
+};
+
+typedef struct scheduled_task_impl scheduled_task;
```

Comment:
I'm renaming it to task_instance now, which is what it really is! We are already using the assign terminology to refer to assigning tasks to workers, so it is not great in the context I think.